packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -412,6 +486,7 @@ class ContainerLayer extends Layer {
     assert(child.attached == attached);
     assert(_debugUltimatePreviousSiblingOf(child, equals: firstChild));
     assert(_debugUltimateNextSiblingOf(child, equals: lastChild));
+    markDirty();
```

Comment:
this should probably be done in Layer.dropChild, not ContainerLayer._removeChild, in case we ever add another kind of layer that has children.